components/ws-manager-bridge/ee/src/prebuild-updater-db.ts

```diff
@@ -78,6 +78,7 @@ export class PrebuildUpdaterDB implements PrebuildUpdater {
 
                 span.setTag("updatePrebuildWorkspace.prebuild.state", updatedPrebuild.state);
                 span.setTag("updatePrebuildWorkspace.prebuild.error", updatedPrebuild.error);
+                this.prometheusExporter.increasePrebuildsCounter(updatedPrebuild.state);
```

Comment:
@geropl I ended up using only one counter because here the updated state can be either starting ("building") or ending.

Reply:
Semantically, a counter is not correct for this. You're tracking state which changes over time. This means that if I do `sum(gitpod_prebuilds_total)` I will receive 1 for each state a prebuild has been in, not a total number of prebuilds.

For this case, you need to use multiple metrics - 1 gauge to track the state. If you also want to know things like # of started prebuilds vs # of succesful prebuilds, you may need extra metrics.

Reply:
Thanks for the comment, @easyCZ! Indeed that for the total number of prebuilds, we will need a gauge. However that means reading the DB and we weren't sure that would be the best way. Having a counter for the amount of each state a prebuild has been in would serve the purpose of comparing how many `queued` and how many ended up in an "end" state. With this we can see how many never got to an end state and debug from there. 
What do you think?

Reply:
Also depending on the desired accuracy, you may have to read from the DB as the DB is the only source of truth. If we are happy with some amount of error, handling this from the ws-manager-bridge is fine (but expect that when ws-manager-bridge fails, or otherwise doesn't work, your metrics will be off)

Reply:
After discussing with @easyCZ, we'll have this PR to add the counter metrics and see how they help us. If in the future we deem a gauge necessary then we will follow-up with it. 
A fix will be made in this PR to separate the counter into two - started and completed. It is now a draft until the fix is pushed.